sympy/core/tests/test_arit.py

```diff
@@ -1501,7 +1508,7 @@ def test_Mul_is_irrational():
     expr = Mul(1, 2, 3, evaluate=False)
     assert expr.is_irrational is False
     expr = Mul(1, I, I, evaluate=False)
-    assert expr.is_rational is True # I * I = -1
+    assert expr.is_rational is None # I * I = -1 but *no evaluation allowed*
```

Comment:
```suggestion
    assert expr.is_rational is not False # I * I = -1 but *no evaluation allowed*
```

Reply:
and this is more appropriate right ?

Reply:
In cases like this I would prefer to assert that is is None but have a comment indicating the ideally it would be True. Then the test shows what is actually returned but also the comment explains why.

Then if the output changes for some reason the test will fail and someone can change it.

Reply:
-1 is Rational...I don't think the change is necessary

Reply:
I think the "count the I" method is in the `is_real` routine; This is such a corner case I am not sure that it is worth the code and the penalty of having to compute it for all the cases when it won't be true: it's an integer if 
```python
i, d = sift(self.args, lambda x: x is S.ImaginaryUnit, binary=True)
if i:
    return fuzzy_and([len(i)%2 == 0, Mul(*d,evaluate=False).is_integer])
```